File: lecture_009(Recursion)/l_003_printdecreasingincreasing.cpp
#include<iostream>
using namespace std;

void printdecreasingincreasing(int n) {
    if(n==0) {
        return;
    }
    cout<<n<<endl;
    printdecreasingincreasing(n-1);
    cout<<n<<endl;
}
int main() {
    int n; 
    cin>>n;
    printdecreasingincreasing(n);

    return 0;
}

File: lecture_009(Recursion)/l_003_printdecreasingincreasing.java
import java.util.Scanner;
public class l_003_printdecreasingincreasing {
    public static Scanner scn = new Scanner(System.in);
    public static void main(String[] args) {
        int n = scn.nextInt();
        printDecreasingIncreasing(n);
    }
    public static void printDecreasingIncreasing(int n) {
        if(n==0) {
            return;
        }
        System.out.println(n);
        printDecreasingIncreasing(n-1);
        System.out.println(n);
    }
}